src/javascript/app/pages/user/account/settings/account_closure.js

```diff
@@ -1,376 +1,167 @@
+            e.preventDefault();
+            showStep(2);
         });
 
-        $(form_selector).on('submit', (event) => {
-            event.preventDefault();
-            if (getReason()) {
-                $account_closure_warning.setVisibility(1);
+        el_step_2_submit.addEventListener('click', (e) => {
+            if (!el_step_2_submit.classList.contains('button-disabled')) {
+                e.preventDefault();
+                el_account_closure_warning.setVisibility(1);
             }
         });
 
-        $txt_other_reason.setVisibility(0);
+        el_step_2_back.addEventListener('click', () => {
+            showStep(1);
+            $.scrollTo(0, 500);
+        });
 
-        $txt_other_reason.on('keyup', () => {
-            const input = $txt_other_reason.val();
-            if (input && validateReasonTextField(false)) {
-                $txt_other_reason.removeClass('error-field');
-                $error_msg.css('display', 'none');
-            }
+        reason_checkbox_list.forEach(element => {
+            element.addEventListener('change', () => { onSelectedReasonChange(); });
         });
-        $('#reason input[type=radio]').on('change', (e) => {
-            const { value } = e.target;
 
-            if (value === 'other') {
-                $txt_other_reason.setVisibility(1);
-            } else {
-                $txt_other_reason.setVisibility(0);
-                $txt_other_reason.removeClass('error-field');
-                $txt_other_reason.val('');
-                $error_msg.css('display', 'none');
-            }
+        el_suggested_improves.addEventListener('input', onTextChanged);
+        el_other_trading_platforms.addEventListener('input', onTextChanged);
+    };
+
+    const showStep = (step) => {
+        Array.from(new Array(3)).forEach((_, index) => {
+            getElementById(`step_${index + 1}`).setVisibility(index + 1 === step);
         });
     };
 
-    const submitForm = ($account_closure_error) => {
-        const $btn_submit = $form.find('#btn_submit');
-        $submit_loading.setVisibility(1);
-        $btn_submit.attr('disabled', true);
+    const onTextChanged = (e) => {
+        const regex = new RegExp('^[a-zA-Z0-9., \'-]+$');
+        if (!regex.test(e.data) ||
+            el_other_trading_platforms.value.length + el_suggested_improves.value.length > 255) {
+            document.execCommand('undo');
+            return;
+        }
+        el_remain_characters.innerHTML = localize(
+            'Remaining characters: [_1].', 255
+            - el_other_trading_platforms.value.length
+        - el_suggested_improves.value.length);
+    };
+
+    const onSelectedReasonChange = () => {
+        const countOfSelectedReasons = getSelectedReasonCount();
+        if (countOfSelectedReasons === 0) {
+            el_step_2_submit.classList.add('button-disabled');
+        } else {
+            el_step_2_submit.classList.remove('button-disabled');
+        }
```

Comment:
my suggestion, up to you

`el_step_2_submit.classList[countOfSelectedReasons ? 'remove' : 'add']('button-disabled');`

Reply:
Very interesting suggestion. Thank you